plugin/evm/vm_test.go

```diff
@@ -770,44 +742,343 @@ func TestIssueAtomicTxs(t *testing.T) {
 	<-issuer
 
 	blk2, err := vm.BuildBlock(context.Background())
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(err)
 
-	if err := blk2.Verify(context.Background()); err != nil {
-		t.Fatal(err)
-	}
+	err = blk2.Verify(context.Background())
+	require.NoError(err)
 
-	if status := blk2.Status(); status != choices.Processing {
-		t.Fatalf("Expected status of built block to be %s, but found %s", choices.Processing, status)
+	require.Equal(blk2.Status(), choices.Processing, "Expected status to match")
+
+	err = blk2.Accept(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk2.Status(), choices.Accepted, "Expected status to match")
+
+	lastAcceptedID, err = vm.LastAccepted(context.Background())
+	require.NoError(err)
+	require.Equal(lastAcceptedID, blk2.ID(), "Expected last accepted blockID to be the accepted block")
+
+	vm.blockChain.DrainAcceptorQueue()
+
+	exportTx2, err := vm.newExportTx(vm.ctx.AVAXAssetID, 1, vm.ctx.XChainID, testShortIDAddrs[0], initialBaseFee, []*secp256k1.PrivateKey{testKeys[0]})
+	require.NoError(err)
+
+	err = vm.issueTx(exportTx2, true /*=local*/)
+	require.NoError(err)
+
+	<-issuer
+
+	blk3, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	err = vm.mempool.ForceAddTx(exportTx2)
```

Comment:
Could we create a separate test for this instead of adding to the existing one? Alternatively, can we get rid of the modification to this test in favor of just adding the regression test below?

Reply:
I originally kept this as a separate test but merged it based on other's feedback as it tests the same function. It should be able to test the accepted case as well as the processing case. Is there a reason that the processing test should be kept separate? I see this as a more complete e2e test whereas the other test just tests different input cases.